Segmentation violation when attempting an Apptainer Checkpoint

### Version of Apptainer

`apptainer version 1.3.0-1.el7`

### Expected behavior

Following the instructions at https://apptainer.org/docs/user/1.3/checkpoint.html I expect to be able to take a container checkpoint when running `apptainer checkpoint instance server`

### Actual behavior

```
$ apptainer checkpoint instance server
apptainer checkpoint instance server
INFO:    Using checkpoint "example-checkpoint"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x563f097126de]

```

### Steps to reproduce this behavior

Just follow the example at https://apptainer.org/docs/user/1.3/checkpoint.html and attempt the creation of a checkpoint. 

### What OS/distro are you running

```sh
$ cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"

```

### How did you install Apptainer

Using the pre-built binaries as explained in "Install unprivileged from pre-built binaries" (https://apptainer.org/docs/admin/1.3/installation.html#install-unprivileged-from-pre-built-binaries)

Thanks for the report.

I was able to reproduce the crash on Ubuntu 22.04 after changing the `From` in server.def to `python:3.10-bookworm`.  With 3.10-buster as it shows in the example, I got
```
usr/bin/dmtcp_launch: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /usr/bin/dmtcp_launch)
/usr/bin/dmtcp_launch: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /usr/bin/dmtcp_launch)
```
on the `apptainer instance start --dmtp-launch...` step.

It also shows a stack trace.  I suppose it did for you too but you didn't include it in your description.  The full output is
```
$ apptainer checkpoint instance server
INFO:    Using checkpoint "example-checkpoint"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x559b32f2a6fe]

goroutine 1 [running]:
github.com/apptainer/apptainer/cmd/internal/cli.getEncryptionMaterial(0xc00057f4f0?)
	github.com/apptainer/apptainer/cmd/internal/cli/build_linux.go:416 +0x11e
github.com/apptainer/apptainer/cmd/internal/cli.launchContainer(0x559b33fb1f00, {0xc000048db0, 0x11}, {0xc00007cff0, 0x5, 0x5}, {0x0, 0x0}, 0xffffffffffffffff)
	github.com/apptainer/apptainer/cmd/internal/cli/actions.go:299 +0x169
github.com/apptainer/apptainer/cmd/internal/cli.glob..func20(0x559b33fb1f00?, {0xc0000c80f0, 0x1?, 0x1?})
	github.com/apptainer/apptainer/cmd/internal/cli/checkpoint.go:170 +0x345
github.com/spf13/cobra.(*Command).execute(0x559b33fb1f00, {0xc000040070, 0x1, 0x1})
	github.com/spf13/cobra@v1.8.0/command.go:987 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x559b33faf100)
	github.com/spf13/cobra@v1.8.0/command.go:1115 +0x425
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	github.com/spf13/cobra@v1.8.0/command.go:1032
github.com/apptainer/apptainer/cmd/internal/cli.ExecuteApptainer()
	github.com/apptainer/apptainer/cmd/internal/cli/apptainer.go:671 +0x26b
main.main()
	github.com/apptainer/apptainer/cmd/apptainer/cli.go:22 +0x36
```

The stack trace shows a problem with more recent code added for unprivileged encryption.  That was added by @JasonYangShadow so I am reassigning this issue to him.  I don't understand how this wasn't caught by the e2e tests because it is running essentially the same example.

Meanwhile, @angel-devicente I believe you could work around this problem by changing the `getEncryptionMaterial` function in build_linux.go to return `nil, nil` if `PEMFlag == nil` or `passphraseFlag == nil`.  I tried that and got further but then the restarted server seems to be getting stuck on something; it doesn't listen on port 8888 and a stop times out and does a hard kill.  Why don't you see if you have the same experience and if so open another issue since that appears to be a different problem.

The instructions were written prior to the 3.0.0 release and suggested using the master branch at the time if 2.x doesn't work.  The current release is 3.0.0, and the instructions should be brought up to date to say to use the latest 3.x.x release.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,7 @@ For older changes see the [archived Singularity change log](https://github.com/a
 - Added libcudadebugger.so to nvliblist.conf to support cuda-gdb in CUDA 12+.
 - Ensure opened/kept file descriptors in stage 1 are not closed during the Go
   garbage collection to avoid "bad file descriptor" errors at startup.
+- Fixed a segmentation violation issue when running Apptainer checkpoint.
 
 ## v1.3.2 - \[2024-05-28\]
 

--- a/cmd/internal/cli/build_linux.go
+++ b/cmd/internal/cli/build_linux.go
@@ -412,6 +412,10 @@ func getEncryptionMaterial(cmd *cobra.Command) (*cryptkey.KeyInfo, error) {
 	pemPathEnv, pemPathEnvOK := os.LookupEnv("APPTAINER_ENCRYPTION_PEM_PATH")
 	pemDataEnv, pemDataEnvOK := os.LookupEnv("APPTAINER_ENCRYPTION_PEM_DATA")
 
+	if PEMFlag == nil || passphraseFlag == nil {
+		return nil, nil
+	}
+
 	// checks for no flags/envvars being set
 	if !(PEMFlag.Changed || pemPathEnvOK || pemDataEnvOK || passphraseFlag.Changed || passphraseEnvOK) {
 		return nil, nil

